tooling/minimize/src/ty.rs

```diff
@@ -112,6 +112,18 @@ impl<'tcx> Ctxt<'tcx> {
         mini_ty
     }
 
+    /// Returns the minirust type of the pointee, given this `ty` is some form of pointer.
+    pub fn translate_pointee_ty_smir(&mut self, ty: smir::Ty, span: rs::Span) -> Type {
+        let ty = smir::internal(self.tcx, ty);
+        let pointee_ty = match ty.kind() {
+            rs::TyKind::Adt(adt_def, _) if adt_def.is_box() => ty.boxed_ty(),
+            rs::TyKind::Ref(_, ty, _mutbl) => *ty,
+            rs::TyKind::RawPtr(ty, _mutbl) => *ty,
+            x => rs::span_bug!(span, "TyKind not a pointer type: {x:?}"),
+        };
```

Comment:
This is just `ty.builtin_deref(true).unwrap()`.

I am not sure if this needs a dedicated helper function.

Reply:
Yeah sure, didn't know about this of course